src/lib.rs

```diff
@@ -556,11 +556,14 @@ P: Point+'static, V: Value+'static {
 
 type Out<P,V> = Option<Result<(P,V,Location),Error>>;
 
-/// Stream of `Result<(Point,Value,Location)>` data returned by `db.query()`.
-pub struct QueryStream<P,V> where P: Point, V: Value {
-  index: usize,
-  queries: Vec<SubStream<P,V>>,
-  deletes: Arc<Mutex<HashSet<Location>>>
+pin_project_lite::pin_project! {
+  /// Stream of `Result<(Point,Value,Location)>` data returned by `db.query()`.
+  pub struct QueryStream<P,V> where P: Point, V: Value {
+    index: usize,
+    #[pin]
+    queries: Vec<SubStream<P,V>>,
```

Comment:
`queries` holds  a Vec of Streams. This is what we need to mark as `pin`; all the other parts are `unpin`. This removes the need for the manual invocations of `pin_utils` (which I was having some trouble with).